web/src/main/java/org/cbioportal/web/DataAccessTokenController.java

```diff
@@ -0,0 +1,123 @@
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.client.HttpClientErrorException;
+
+@InternalApi
+@RestController
+@Validated
+@Api(tags = "Data Access Tokens", description = " ")
+public class DataAccessTokenController {
+
+    @Autowired
+    // use @Qualifier to ensure we get tokenService bean from applicationContext-security.xml
+    // tokenSerice bean in security.xml file has same name so would be picked anyway by default,
+    // but this avoids a NoUniqueBeanDefinitionException
+    @Qualifier("tokenService")
+    private DataAccessTokenService dataAccessTokenService;
+
+//      TODO: figure out how to read this from the properties file
+//    @Value("${security.data_tokens.unauth_users}")
+    private String[] USERS_WHO_CANNOT_USE_TOKENS = {"anonymousUser", "servcbioportal"};
+    private Set<String> usersWhoCannotUseTokenSet = null;
+
+    @Autowired
+    private void initializeUsersWhoCannotUseTokenSet() {
+        usersWhoCannotUseTokenSet = new HashSet<>(Arrays.asList(USERS_WHO_CANNOT_USE_TOKENS));
+    }
+
+    @RequestMapping(method = RequestMethod.POST, value = "/data-access-tokens", produces = MediaType.APPLICATION_JSON_VALUE)
+    public ResponseEntity<DataAccessToken> createDataAccessToken(Authentication authentication,
+                                    @RequestParam(required = false) Boolean allowRevocationOfOtherTokens) throws HttpClientErrorException {
+        DataAccessToken createdToken;
+        if (allowRevocationOfOtherTokens != null) {
+            createdToken = dataAccessTokenService.createDataAccessToken(getAuthenticatedUser(authentication), allowRevocationOfOtherTokens);
+        }
+        else {
+            createdToken = dataAccessTokenService.createDataAccessToken(getAuthenticatedUser(authentication));
+        }
+        if (createdToken == null) {
+            return new ResponseEntity<>(new DataAccessToken(null), HttpStatus.NOT_FOUND);
+        }
+        return new ResponseEntity<>(createdToken, HttpStatus.CREATED);
+    }
+
+    @RequestMapping(method = RequestMethod.GET, value = "/data-access-tokens/fetch")
+    public ResponseEntity<List<DataAccessToken>> getAllDataAccessTokens(Authentication authentication) {
+        List<DataAccessToken> allDataAccessTokens = dataAccessTokenService.getAllDataAccessTokens(getAuthenticatedUser(authentication));
+        return new ResponseEntity<>(allDataAccessTokens, HttpStatus.OK);
+    }
+
+    @RequestMapping(method = RequestMethod.GET, value = "/data-access-tokens/{token}")
+    public ResponseEntity<DataAccessToken> getDataAccessToken(
+            @ApiParam(required = true, value = "token") @PathVariable String token) {
+        DataAccessToken dataAccessToken = dataAccessTokenService.getDataAccessTokenInfo(token);
+        return new ResponseEntity<>(dataAccessToken, HttpStatus.OK);
+    }
+
+    @RequestMapping(method = RequestMethod.DELETE, value = "/data-access-tokens/revoke")
```

Comment:
You don't need `/revoke` either, those verbs at the end of the paths is for POST requests that does querying (when it's a complex or long query that can't be done with GET parameters), to distinguish them from the POST requests that creates/updates resources.